### PR TITLE
Test Network Policies

### PR DESCRIPTION
**Pull request by mateoflorido**
## Changes
- Add tests for Network Policies in Kube-OVN.

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -125,14 +125,27 @@ def gateway_client_pod(client, worker_node, subnet_resource):
...
 
-async def wait_pod_ip(client, pod):
-    for _, obj in client.watch(
-        Pod,
-        namespace=pod.metadata.namespace,
-        fields={"metadata.name": pod.metadata.name},
-    ):
-        if obj.status.podIP:
-            return obj
+async def wait_pod_ips(client, pods):
```
Comment (addyess) [inline tests/integration/conftest.py:128]: how about some docstring here to help indicate what this method does

```suggestion
async def wait_pod_ips(client, pods):
    """Returns a list of pods which have an ip address assigned."""

```

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -154,28 +167,23 @@ async def isolated_subnet(client, subnet_resource):
...
     yield tuple(pods)
 
     log.info("Deleting isolated subnet resources ...")
     for obj in codecs.load_all_yaml(path.read_text()):
         client.delete(type(obj), obj.metadata.name, namespace=obj.metadata.namespace)
 
+    await wait_for_removal(client, pods)
+
+
+async def wait_for_removal(client, pods):
```
Comment (addyess) [inline tests/integration/conftest.py:181]: how about some docstring here to help indicate what this method does

```suggestion
async def wait_for_removal(client, pods):
    """Waits until listed pods are no longer present in the cluster."""
```

File: tests/integration/test_kube_ovn.py [hunk 3]
```diff
@@ -445,6 +447,33 @@ async def test_bgp(ops_test, ips_to_curl):
...
+        assert msg in stdout
+
+    log.info("Checking pods connectivity...")
+    for pod in network_policies:
+        await check_wget("nginx.netpolicy", pod, "'index.html' saved")
+
+    log.info("Applying NetworkPolicy...")
+    path = Path("tests/data/net-policy.yaml")
+    for obj in load_all_yaml(path.read_text()):
+        client.create(obj)
```
Comment (addyess) [inline tests/integration/test_kube_ovn.py:470]: does this need cleaning up/removal?

**Review by addyess — APPROVED** [review]
I have questions but this looks like it tests something 😃

File: tests/integration/conftest.py (hunk 1 above)
Comment (mateoflorido) [reply tests/integration/conftest.py:128]: You're dead right! I'll add both docstrings to the methods.

File: tests/integration/test_kube_ovn.py [hunk 4]
```diff
@@ -445,6 +447,38 @@ async def test_bgp(ops_test, ips_to_curl):
...
+    for obj in policies:
+        client.create(obj)
+
+    log.info("Checking NetworkPolicy...")
+    await check_wget("nginx.netpolicy", allowed_pod, "'index.html' saved")
+    await check_wget("nginx.netpolicy", blocked_pod, "wget: download timed out")
+
+    log.info("Removing NetworkPolicy...")
+    for obj in policies:
+        client.delete(type(obj), obj.metadata.name, obj.metadata.namespace)
```
Comment (addyess) [inline tests/integration/test_kube_ovn.py:479]: may i suggest this alternative as a way to still teardown even if the checks in the `try` block fail

```suggestion
    try:
        log.info("Checking NetworkPolicy...")
        await check_wget("nginx.netpolicy", allowed_pod, "'index.html' saved")
        await check_wget("nginx.netpolicy", blocked_pod, "wget: download timed out")
    finally:
        log.info("Removing NetworkPolicy...")
        for obj in policies:
            client.delete(type(obj), obj.metadata.name, obj.metadata.namespace)
```

**Review by addyess — CHANGES_REQUESTED** [review]
new little nit

File: tests/integration/test_kube_ovn.py (hunk 4 above)
Comment (mateoflorido) [reply tests/integration/test_kube_ovn.py]: Thanks @addyess, I will commit your suggestion straight away 😊

**Review by addyess — APPROVED** [review]
NICE!  🚀

**Pull request merged**